### DISCUSSION TITLE
Zoom and MS Teams Integration?

### DISCUSSION THREAD

**Discussion by leav6** [iXab3r/MicSwitch#64] [Ideas]
Hi Dimitry, 
I really like the versatility and customizability of MicSwitch!

Would it be possible to synchronize the MicSwitch actions with web conferencing software?
Being able to reduce or eliminate the dreaded "double-muted" situation (microphone is unmuted but webinar app microphone mute is still on) would be a relief.

**Comment by leav6** [comment]
I found a partial workaround. 
I can enable audio mute/unmute keyboard shortcut (hotkey) in Zoom, using the same one as configured in MicSwitch. 

The issue seems to be inconsistent reaction when pressing the hotkey (I selected the Pause Break key), i.e. when Zoom toggles its mute off, MicSwitch toggles to mute on. 

When in this opposite (un)mute situation, I can click the Zoom or MicSwitch microphone toggle to bring them in sync. 
When the hotkey unmutes Zoom, MicSwitch mutes. Then subsequent hotkey presses toggle Zoom and MicSwitch so they remain out of sync.

**Reply by leav6** [reply]
I'll investigate if this helps prevent the out of sync toggle. 
https://answers.microsoft.com/en-us/windows/forum/all/auto-muting-of-sound-and-microphone-at-random/e52de41b-f666-463c-a5b5-bfae422dedae

**Reply by leav6** [reply]
An example of where the control is happening to Zoom and Teams is with the Jabra Speak 510.
This speakerphone controls the software without shortcut keys or other non-default settings in the web conference settings.

**Reply by leav6** [reply]
For now, I will leave Zoom unmuted and mute/unmute in MicSwitch.

**Comment by iXab3r** [comment]
Hi ! I am using the same approach in all programs that interact with microphone - leave them unmuted an mute/unmute via micswitch. Trying to synchronize state in all programs that interact with microphone is possible, but it will require a lot of work and solution will be very error-prone anyways, don't see myself implementing it in foreseeble future

**Reply by leav6** [reply]
Thanks, that makes sense.
